Sources/Orbit/Components/SegmentedSwitch.swift

```diff
@@ -0,0 +1,157 @@
+import SwiftUI
+
+/// Showing two choices from which only one can be selected.
+///
+/// Currently SegmentedSwitch allows to have 2 segments.
+/// SegmentedSwitch can be in error state only in unselected state.
+///
+/// - Note: [Orbit definition](https://orbit.kiwi/components/interaction/segmentedswitch/)
+public struct SegmentedSwitch<Value>: View where Value: LosslessStringConvertible & Equatable {
+
+    @Binding private var selection: Value?
+    let label: String
+    let _message: Message?
+    let firstOption: Value
+    let secondOption: Value
+
+    let separatorWidth: CGFloat = .xxxSmall
+    let borderWidth: CGFloat = .xxxSmall
+
+    public var body: some View {
+        FieldWrapper(
+            label,
+            message: message,
+            content: {
+                HStack(spacing: 0) {
+                    segment(title: firstOption)
+                    separator
+                    segment(title: secondOption)
+                }
+                .background(selectionBackground)
+                .background(roundedBackground)
+                .fixedSize(horizontal: false, vertical: true)
+            }
+        )
+    }
+
+    @ViewBuilder func segment(title: Value) -> some View {
+        Text(title.description)
+            .lineLimit(1)
+            .padding(.vertical, .xSmall)
+            .frame(maxWidth: .infinity)
+            .contentShape(Rectangle())
+            .onTapGesture {
+                selection = title
+            }
+    }
+
+    @ViewBuilder var separator: some View {
+        Rectangle()
+            .fill(borderColor)
+            .frame(width: separatorWidth)
+    }
+
+    @ViewBuilder var selectionBackground: some View {
+        GeometryReader { reader in
+            if selection != nil {
+                Color.whiteDarker
+                    .frame(width: (reader.size.width - 2 * borderWidth) / 2)
+                    .clipShape(RoundedRectangle(cornerRadius: BorderRadius.default))
+                    .padding(borderWidth)
+                    .elevation(.level1)
+                    .offset(computeSelectionOffset(in: reader))
+                    .animation(.easeOut(duration: 0.2), value: selection)
+            }
+        }
+    }
+
+    @ViewBuilder var roundedBackground: some View {
+        RoundedRectangle(cornerRadius: BorderRadius.default)
+            .fill(backgroundColor)
+            .overlay(
+                RoundedRectangle(cornerRadius: BorderRadius.default)
+                    .stroke(borderColor, lineWidth: borderWidth)
+            )
+    }
+
+    private var message: Message? {
+        if case .error = _message, selection != nil {
```

Comment:
Would this code be longer as a `switch`? If not, please use that.

Reply:
Or, can we simply use the `message` from call-site, as is? We should show if client provides it

Reply:
we agreed that error msg can be used only when there is no selection.